tests/models/test_connection.py

```diff
@@ -118,6 +118,39 @@ class TestConnection:
                 None,
                 r"Invalid connection string: type://user:pass@protocol://host:port?param=value.",
             ),
+            (
+                "postgres://jdbc:postgresql://10.0.0.1",
+                "postgres",
+                "jdbc:postgresql://10.0.0.1",
+                None,
+                None,
+                None,
+                "",
+                {},
+                r"Invalid connection string: postgres://jdbc:postgresql://10.0.0.1.",
+            ),
+            (
+                "jdbc:postgresql://user:host@10.0.0.1/database",
+                "jdbc",
+                "postgresql://user:host@10.0.0.1/database",
```

Comment:
IMO for Airflow connection uri, we should only accept `<conn type>://<conn url>` (and remove all the connections which doesn't have a type in the unit tests or add a type by default like `default://<conn url>`).

There is no reason to consider the jdbc as conn type and schema at the same time in the provided uri.

Reply:
I **think** I solved it a bit more complete now.... Please take a look.

Reply:
And ... We have a back-compatibility problem here, I think we can't just "drop" those other "URI-like" ways of specifying the connection, I am afraid.